cmd/setup-etcd-environment/run.go

```diff
@@ -129,53 +164,82 @@ func runRunCmd(cmd *cobra.Command, args []string) error {
 		out = f
 	}
 
-	if runOpts.bootstrapSRV {
-		exportEnv["DISCOVERY_SRV"] = runOpts.discoverySRV
-	} else {
-		exportEnv["NAME"] = etcdName
+	if err := writeEnvironmentFile(exportEnv, out, true); err != nil {
+		return err
 	}
 
-	// enable etcd to run using s390 and s390x. Because these are not officially supported upstream
-	// etcd requires population of environment variable ETCD_UNSUPPORTED_ARCH at runtime.
-	// https://github.com/etcd-io/etcd/blob/master/Documentation/op-guide/supported-platform.md
-	arch := runtime.GOARCH
-	if strings.HasPrefix(arch, "s390") {
-		exportEnv["UNSUPPORTED_ARCH"] = arch
+	noExportEnv := map[string]string{
+		"IPV4_ADDRESS":      setupEnv.etcdIP,
+		"WILDCARD_DNS_NAME": fmt.Sprintf("*.%s", setupEnv.opts.discoverySRV),
 	}
-	if err := writeEnvironmentFile(exportEnv, out, true); err != nil {
-		return err
+	if setupEnv.etcdDNS != "" {
+		noExportEnv["DNS_NAME"] = setupEnv.etcdDNS
 	}
+	return writeEnvironmentFile(noExportEnv, out, false)
+}
 
-	return writeEnvironmentFile(map[string]string{
-		"IPV4_ADDRESS":      ip,
-		"DNS_NAME":          dns,
-		"WILDCARD_DNS_NAME": fmt.Sprintf("*.%s", runOpts.discoverySRV),
-	}, out, false)
+func (s *SetupEnv) getExportEnv() (map[string]string, error) {
+	if s.isEtcdScaling() {
+		return s.getOperatorEnv()
+	}
+	// initialize envs used to bootstrap etcd
+	return s.getBootstrapEnv()
 }
 
-func setExportEnv(etcdName, dns string) (map[string]string, error) {
-	exportEnv := make(map[string]string)
-	duration := 10 * time.Second
-	wait.PollInfinite(duration, func() (bool, error) {
-		if _, err := os.Stat("/var/run/secrets/kubernetes.io/serviceaccount/token"); os.IsNotExist(err) {
-			glog.Errorf("serviceaccount failed: %v", err)
-			return false, nil
-		}
-		return true, nil
-	})
+// getOperatorEnv returns a maping of ENV variables required to scale etcd via operator
+func (s *SetupEnv) getOperatorEnv() (map[string]string, error) {
+	env := make(map[string]string)
+	env["NAME"] = s.etcdName
+
+	memberList := s.getEtcdMemberList()
+	env["INITIAL_CLUSTER"] = strings.Join(memberList, ",")
+	env["INITIAL_CLUSTER_STATE"] = etcdInitialExisting
+
+	endpoints, err := s.getEtcdEndpoints()
```

Comment:
We probably need to have something here that verifies the member list we got [here](https://github.com/openshift/machine-config-operator/pull/1287/files#diff-f26a1d5dc0cdd6c1f8863a9a3d7a3630R194) with the endpoint list and prove a one on one correspondence. 

Can we have a TODO here and corresponding issue/trello card to track it for a follow-up PR?

Reply:
>We probably need to have something here that verifies the member list we got here with the endpoint list and prove a one on one correspondence.

Why?

If this list[1] is good enough for apiserver it better be good enough for us. we have a client balancer to handle failover if a member is not valid/down etc. If the list were not the same would we fail to bootstrap? I think this type of validation belongs in the operator. 

[1] https://github.com/openshift/machine-config-operator/pull/1287/files#diff-f26a1d5dc0cdd6c1f8863a9a3d7a3630R211

Reply:
I see your point that it is important if we have a bug but we need to guarantee the list is valid before we ship it IMO and that is on the operator.

Reply:
We have changes coming to the operator to allow this validation we can talk about that later.

Reply:
I agree the validation should be on the cluster-etcd-operator. Feel free to mark this resolved